### PR TITLE
Use meilisearch instead of redisearch

### PR DESCRIPTION
**Pull request by razzeee**
Pretty happy with this so far. Search returns might still be able to be improved, but so far we get a bunch of stuff for free. Not sure how good sorting by downloads works right now, as it's not the highest ranked value (which we probably don't want anyway)

`news` shows stuff that's along those lines + words that are similar
Do `"news"` for a more literal match.

You can also access a nice interface when running this via localhost:7700 to check out the search index

I'll also look at https://github.com/meilisearch/meilisearch-react/ when this is done.

### CODE REVIEW

File: app/apps.py [hunk 1]
```diff
@@ -1,60 +1,65 @@
...
-            }
+            search_keywords = apps[appid].get("keywords")
+
+            # order of the dict is important for attritbute ranking
+            search_apps.append(
+                {
+                    "id": utils.get_clean_app_id(appid),
+                    "name": apps[appid]["name"],
+                    "summary": apps[appid]["summary"],
+                    "downloads_last_month": app_downloads.get(appid, {'downloads_last_month': 0}).get('downloads_last_month', 0),
```
Comment (razzeee) [inline app/apps.py:33]: `app_downloads` has everything we have stats for, but we only want items, that are in the appstream to get pushed to the search (e.g. no EOL apps)

File: docker-compose.yml [hunk 2]
```diff
@@ -1,11 +1,17 @@
 version: "3"
 
 services:
+  # replace with normal redis?
   redis:
     image: docker.io/redislabs/redisearch:2.4.0
     ports:
       - "6379:6379"
 
+  meilisearch:
```
Comment (razzeee) [inline docker-compose.yml:10]: We need to make sure, that this service is not exposed to the web in the staging/production deployments

File: docker-compose.yml [hunk 3]
```diff
@@ -1,11 +1,17 @@
 version: "3"
 
 services:
+  # replace with normal redis?
```
Comment (razzeee) [inline docker-compose.yml:4]: Not sure if we want to do this at all or if it should be a different MR

**Comment by barthalion** [comment]
> I'll also look at https://github.com/meilisearch/meilisearch-react/ when this is done.

Does this mean mellisearch would need to be exposed publicly? I would rather not expose it even if it's just for search.

File: docker-compose.yml (hunk 3 above)
Comment (barthalion) [reply docker-compose.yml:4]: We can do it here as it's used only for local development.

File: app/main.py [hunk 4]
```diff
@@ -43,17 +44,16 @@
...
 
-    db.initialize()
     picks.initialize()
     verification.initialize()
 
 
 @app.post("/update")
 def update():
-    new_apps = apps.load_appstream()
+    app_downloads = stats.update()
```
Comment (barthalion) [inline app/main.py:53]: Not sure if I follow what happens here?

Comment (razzeee) [reply app/main.py:53]: Running stats updates first, so that we can generate a list of appId + downloads last 30 days

Then we can just use that list when we do `load_appstream`. Stats has `EOL` apps and so does our download list. But as we run `load_appstream`, we can just access our key/values via an `appId` from the appstream. So this makes sure we don't create search entries for apps that are not in the appstream.

**Comment by razzeee** [comment]
> Does this mean mellisearch would need to be exposed publicly? I would rather not expose it even if it's just for search.

Your probably right. That would be a prerequisite. Meilisearch can be setup to use a master_key, so it should be guarded, but totally fine, if you don't feel comfortable with that.

File: app/main.py (hunk 4 above)
Comment (razzeee) [reply app/main.py]: Had to revert this, as stats already depend on data from the appstream, as it accesses the database to figure out how many apps there are.

File: docker-compose.yml (hunk 3 above)
Comment (razzeee) [reply docker-compose.yml]: I've removed the comment, can't figure out why it's not working with the normal redis image

**Comment by razzeee** [comment]
The react instant search seems to work nicely, but it's probably more geared towards a full page search like https://codesandbox.io/s/ms-react-is-sh9ud?fontsize=14&hidenavigation=1&theme=dark and not a search bar like we have. So it's probably nothing we want.

**Comment by irevoire** [comment]
Yes @razzeee is right, if you set a master key you can expose meilisearch without any issue.
And if you want to remove the dashboard that meilisearch generate on localhost:7700 by default you can start it with `--env production` and it'll be disabled 👍

**Comment by razzeee** [comment]
Moved to https://github.com/flathub/website/pull/64
